smoke-tests/src/androidTestFirestore/java/com/google/firebase/testing/firestore/FirestoreTest.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.testing.firestore;
+
+import android.util.Log;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+import com.google.firebase.testing.common.SmokeTestBase;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public final class FirestoreTest extends SmokeTestBase {
```

Comment:
Prefer composition over inheritance as over time the latter produces deep rigid inheritance hierarchies that are hard to maintain and reason about. So I'd suggest you have a `SmokeTestRule` instead as that would be much more composeable.

Alternatively, it looks like `SmokeTestBase` is stateless so `runSmokeTests` can be static and be put in a Util class, so no inheritance or rule would be required. wdyt?

Reply:
Done. I moved it to SmokeTestRunner.